test/endpoint_provisioner_test.go

```diff
@@ -0,0 +1,1083 @@
+				clientResponses, _, err := testcontext.SEMP().Monitor().MsgVpnApi.
+					GetMsgVpnQueues(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, nil)
+				Expect(err).To(BeNil())
+				Expect(clientResponses.Data).ToNot(BeNil())
+				Expect(len(clientResponses.Data)).To(Equal(0)) // should have no queues
+			})
+		}
+
+		// invalid queue names to test
+		invalidQueueTestCases := map[string]string{
+			"NameLongerThan201CharactersQueueName": "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ",
+			"ReplayLogQueueName":                   "#REPLAY_LOG_defaultLog",
+			"ReplayTopicsQueueName":                "#REPLAY_TOPICS_defaultLog",
+			"PartitionQueuePrefixQueueName":        "#pq/db583a27835baecd/00011",
+			"TopicLevelsQueueName":                 "/test1/test2/",
+		}
+		for queueLabel, queueName := range invalidQueueTestCases {
+			// TestPlan TestCase Provision#1
+			It("should not provision queue with invalid queueName - "+queueLabel, func() {
+				// remove the provisioned queue
+				defer func() {
+					err := provisioner.Deprovision(queueName, true)
+					Expect(err).ToNot(BeNil())
+					Expect(string(err.Error())).To(Equal("Invalid Queue Name"))
+				}()
+
+				// check that the endpoint does not yet exist on the broker via semp
+				clientResponse, _, err := testcontext.SEMP().Monitor().MsgVpnApi.
+					GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(queueName), nil)
+				Expect(err).ToNot(BeNil())
+
+				// Now let's provision the endpoint using the API
+				provisioner = provisioner.FromConfigurationProvider(config.EndpointPropertyMap{
+					config.EndpointPropertyDurable:   true,
+					config.EndpointPropertyExclusive: false,
+				})
+				outcome := provisioner.Provision(queueName, false)
+				Expect(outcome.GetStatus()).To(BeFalse())
+				Expect(outcome.GetError()).To(HaveOccurred())
+				Expect(string(outcome.GetError().Error())).To(Equal("Invalid Queue Name"))
+
+				// check that the endpoint was provisioned on the broker via semp
+				clientResponse, _, err = testcontext.SEMP().Monitor().MsgVpnApi.
+					GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(queueName), nil)
+				Expect(err).ToNot(BeNil())
+				Expect(clientResponse.Data).To(BeNil())
+			})
+		}
+
+		// valid queue names to test
+		validQueueTestCases := map[string]string{
+			"ReplicationDataQueueName": "#MSGVPN_REPLICATION_DATA_QUEUE",
+			"CfgSyncQueueName":         "#CFGSYNC/OWNER/vmr-135-47/RTR/site/CFG",
+			"ZeroQueueName":            "00000000000",
+		}
+
+		for queueLabel, queueName := range validQueueTestCases {
+			// TestPlan TestCase Provision#1
+			It("should provision queue with queueName of max length, special characters, numbers only and ancillary type - "+queueLabel, func() {
```

Comment:
Max length is not one of the options in the `validQueueTestCases` map above

Reply:
I was following the Test plan document by QA - https://sol-jira.atlassian.net/wiki/spaces/CPD/pages/4031283346/API+s+Support+For+Removing+Durable+Endpoints+Test+Plan

Reply:
The test name says "queueName of max length" so that's why I was expecting to see a max length queue name

Reply:
I have updated the name @sdewilde